imagecraft/pack/image.py

```diff
@@ -84,7 +84,8 @@ def boot_partition_number(self) -> int | None:
                 return i + 1
         return None
 
-    def attach_loopdev(self) -> str:
+    @contextlib.contextmanager
+    def attach_loopdev(self) -> Generator[str]:
```

Comment:
I think we can return an `Iterator[str]` here instead, since we're yielding just to create the context manager.

Reply:
Good catch. I looked into it because I was not aware of the difference.